src/CoreBundle/private/library/classes/TCMSTableEditor/TCMSTableEditorCMSConfig.class.php

```diff
@@ -65,16 +69,34 @@ public function UpdateTranslationFields()
     protected function PostSaveHook(&$oFields, &$oPostTable)
     {
         parent::PostSaveHook($oFields, $oPostTable);
+
         /**
-         * @var TdbCmsConfig $config
+         * @var $maintenanceModeService MaintenanceModeServiceInterface
          */
-        $config = $this->oTable;
-        if ($config->fieldShutdownWebsites) {
-            touch(PATH_MAINTENANCE_MODE_MARKER);
-        } else {
-            if (file_exists(PATH_MAINTENANCE_MODE_MARKER)) {
-                unlink(PATH_MAINTENANCE_MODE_MARKER);
+        $maintenanceModeService = \ChameleonSystem\CoreBundle\ServiceLocator::get('chameleon_system_core.maintenance_mode_service');
```

Comment:
Could you outsource the service locator call into another method? This moves the ugly call to where we don't see it and we can add the typehint there to further simplify the "real" code.

Reply:
ok